fhirflat/resources/base.py

```diff
@@ -73,6 +77,67 @@ def from_flat(cls, file: str) -> FHIRFlatBase | list[FHIRFlatBase]:
         else:
             return list(df["fhir"])
 
+    @classmethod
+    def ingest_backbone_elements(cls, mapped_data: pd.Series):
+        """
+        Takes ordered lists of data and forms the correct FHIR format which won't
+        be flattened after ingestion.
+        """
+
+        def fhir_format(row):
+            for b_e, b_c in cls.backbone_elements.items():
+                keys_present = [key for key in row if key.startswith(b_e)]
+                if keys_present:
+                    condensed_dict = {k: row[k] for k in keys_present}
+                    if all(
+                        not isinstance(v, list) or len(v) == 1
+                        for v in condensed_dict.values()
+                    ):
+                        continue
+                    else:
+                        backbone_list = []
+                        for i in range(len(next(iter(condensed_dict.values())))):
+                            first_item = {
+                                k.lstrip(b_e + "."): v[i]
+                                for k, v in condensed_dict.items()
+                            }
+                            backbone_list.append(expand_concepts(first_item, b_c))
+                        for k_d in condensed_dict:
+                            row.pop(k_d)
+                        row[b_e] = backbone_list
+            return row
+
+        mapped_data.apply(fhir_format)
```

Comment:
should this be `mapped_data = mapped_data.apply(...)` or does apply work in-place?

Reply:
It works both ways - I'll map it to a new variable for clarity.